src/affexpr.jl

```diff
@@ -15,23 +15,84 @@
 # Operator overloads in src/operators.jl
 #############################################################################
 
+# Utilities for OrderedDict
+function add_or_set!(dict::OrderedDict{K,V}, k::K, v::V) where {K,V}
+    # TODO: This unnecessarily requires two lookups for k.
+    # TODO: Decide if we want to drop zeros here after understanding the
+    # performance implications.
+    dict[k] = get!(dict, k, zero(V)) + v
+    return dict
+end
+
+function new_ordered_dict(::Type{K}, ::Type{V}, kv::AbstractArray{<:Pair}) where {K,V}
+    dict = OrderedDict{K,V}()
+    sizehint!(dict, length(kv))
+    for pair in kv
+        add_or_set!(dict, convert(K, pair.first), convert(V, pair.second))
+    end
+    return dict
+end
+
+function new_ordered_dict(::Type{K}, ::Type{V}, kv::Pair...) where {K,V}
+    dict = OrderedDict{K,V}()
+    sizehint!(dict, length(kv))
+    for pair in kv
+        add_or_set!(dict, convert(K, pair.first), convert(V, pair.second))
+    end
+    return dict
+end
+
+
+
 #############################################################################
 # GenericAffExpr
 # ∑ aᵢ xᵢ  +  c
 mutable struct GenericAffExpr{CoefType,VarType} <: AbstractJuMPScalar
-    vars::Vector{VarType}
-    coeffs::Vector{CoefType}
     constant::CoefType
+    terms::OrderedDict{VarType,CoefType}
```

Comment:
Having `terms` as the first field would be more consistent with `MOI.ScalarAffineFunction`

Reply:
I'm aware of the inconsistency with `MOI.ScalarAffineFunction`, but I like the `AffExpr(constant, pairs...)` constructor, and this ordering is consistent with that.